middle_end/flambda2/terms/code_metadata.ml

```diff
@@ -160,17 +153,17 @@ let createk k code_id ~newer_version_of ~params_arity ~num_trailing_local_params
     ()
   | true, (Always_inline | Unroll _) ->
     Misc.fatal_error "Stubs may not be annotated as [Always_inline] or [Unroll]");
-  if num_trailing_local_params < 0
-     || num_trailing_local_params > Flambda_arity.cardinal params_arity
+  if first_complex_local_param < 0
+     || first_complex_local_param > Flambda_arity.cardinal params_arity
```

Comment:
It both seems fine and kind of weird that `first_complex_local_param = Flambda_arity.cardinal params_arity` is allowed.

Reply:
I added a comment about this as well in the `.mli`.